src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -190,16 +190,82 @@ class AtomCenteredFunction(ArchiveSection):
     Specifies a single function (term) in an atom-centered basis set.
     """
 
-    pass
+    function_type = Quantity(
+        type=MEnum('S', 'P', 'D', 'F', 'G', 'H', 'I', 'J'),
+        description="""
+        the angular momentum of the shell to be added.
+        """,
+    )
+
+    n_primitive = Quantity(
+        type=int,
+        description="""
+        Number of primitives.
+        """,
+    )
+
+    exponents = Quantity(
+        type=np.float32,
+        shape=['n_primitive'],
+        description="""
+        List of exponents for the basis function.
+        """,
+    )
 
-    # TODO: design system for writing basis functions like gaussian or slater orbitals
+    contraction_coefficients = Quantity(
+        type=np.float32,
+        shape=['n_primitive'],
+        description="""
+        List of contraction coefficients corresponding to the exponents.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+        # self.name = self.m_def.name
 
 
 class AtomCenteredBasisSet(BasisSetComponent):
     """
     Defines an atom-centered basis set.
     """
 
+    main_basis_set = Quantity(
+        type=str,
+        description="""
+        Name of the main basis set.
+        """,
+    )
+
+    aux_c_basis_set = Quantity(
+        type=str,
+        description="""
+        AuxC type of basis set.
+        """,
+    )
+
+    aux_j_basis_set = Quantity(
+        type=str,
+        description="""
+        AuxJ type of basis set.
+        """,
+    )
+
+    aux_jk_basis_set = Quantity(
+        type=str,
+        description="""
+        AuxJK type of basis set.
+        """,
+    )
+
+    atoms_ref = Quantity(
+        type=AtomsState,
+        shape=['*'],
+        description="""
+        References to the `AtomsState` sections that define the atoms this basis set applies to.
```

Comment:
Connected to my comment in the issue, you might also want a map to a `ModelSystem`.
For simplicity sake, I'd try to use only 1 type of linker when possible. Convenience can be added in other ways.